### PR TITLE
Removing Miner::default.

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

**Comment by rphmeier** [comment]
the motivation is that Miner::default() was just a test implementation?

File: ethcore/src/json_tests/chain.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ pub fn json_chain_test(json_data: &[u8], era: ChainEra) -> Vec<String> {
 
 			let temp = RandomTempPath::new();
 			{
-				let client = Client::new(ClientConfig::default(), spec, temp.as_path(), Arc::new(Miner::default()), IoChannel::disconnected()).unwrap();
+				let client = Client::new(ClientConfig::default(), spec.clone(), temp.as_path(), Arc::new(Miner::with_spec(spec)), IoChannel::disconnected()).unwrap();
```
Comment (rphmeier) [inline ethcore/src/json_tests/chain.rs]: test build failing because spec can't be cloned

**Comment by tomusdrw** [comment]
Yes, it was just used for tests and it was also misleading. You have to decide which spec should be used (or at least you should be aware that it's spec-dependent).

I've alos removed other constructors, cause we had `Miner::default()`, `Miner::with_accounts(acc)` and `Miner::new(false, spec)`

**Comment by gavofyork** [comment]
doesn't build

**Comment by rphmeier** [comment]
looks like the travis build got cancelled in the run-up to 1.2. I restarted it.

Appveyor failed due to a cargo/git error? I don't have permissions to restart.

**Pull request merged**